types/node/v6/index.d.ts

```diff
@@ -9,4457 +9,40 @@
 *               Node.js v6.x API                *
 *                                               *
 ************************************************/
+// NOTE: These definitions support NodeJS and TypeScript 3.1.
 
-// This needs to be global to avoid TS2403 in case lib.dom.d.ts is present in the same build
-interface Console {
-    Console: typeof NodeJS.Console;
-    assert(value: any, message?: string, ...optionalParams: any[]): void;
-    dir(obj: any, options?: NodeJS.InspectOptions): void;
-    error(message?: any, ...optionalParams: any[]): void;
-    info(message?: any, ...optionalParams: any[]): void;
-    log(message?: any, ...optionalParams: any[]): void;
-    time(label: string): void;
-    timeEnd(label: string): void;
-    trace(message?: any, ...optionalParams: any[]): void;
-    warn(message?: any, ...optionalParams: any[]): void;
-}
+// NOTE: TypeScript version-specific augmentations can be found in the following paths:
+//          - ~/base.d.ts         - Shared definitions common to all TypeScript versions
+//          - ~/index.d.ts        - Definitions specific to TypeScript 2.1
+//          - ~/ts3.1/index.d.ts  - Definitions specific to TypeScript 3.1
 
-interface Error {
-    stack?: string;
-}
+// NOTE: Augmentations for TypeScript 3.1 and later should use individual files for overrides
+//       within the respective ~/ts3.1 (or later) folder. However, this is disallowed for versions
+//       prior to TypeScript 3.1, so the older definitions will be found here.
 
-// Declare "static" methods in Error
-interface ErrorConstructor {
-    /** Create .stack property on a target object */
-    captureStackTrace(targetObject: Object, constructorOpt?: Function): void;
+// Base definitions for all NodeJS modules that are not specific to any version of TypeScript:
+/// <reference path="base.d.ts" />
 
-    /**
-     * Optional override for formatting stack traces
-     *
-     * @see https://github.com/v8/v8/wiki/Stack%20Trace%20API#customizing-stack-traces
-     */
-    prepareStackTrace?: (err: Error, stackTraces: NodeJS.CallSite[]) => any;
+// TypeScript 2.1-specific augmentations:
 
-    stackTraceLimit: number;
-}
-
-// compat for TypeScript 1.8
-// if you use with --target es3 or --target es5 and use below definitions,
-// use the lib.es6.d.ts that is bundled with TypeScript 1.8.
+// Forward-declarations for needed types from es2015 and later (in case users are using `--lib es5`)
 interface MapConstructor { }
 interface WeakMapConstructor { }
 interface SetConstructor { }
 interface WeakSetConstructor { }
-
-// Forward-declare needed types from lib.es2015.d.ts (in case users are using `--lib es5`)
 interface Iterable<T> { }
 interface Iterator<T> {
     next(value?: any): IteratorResult<T>;
 }
-
+interface IterableIterator<T> { }
 interface IteratorResult<T> { }
 interface AsyncIterableIterator<T> {}
 interface SymbolConstructor {
     readonly iterator: symbol;
-    readonly asyncIterator: symbol;
```

Comment:
have you removed `asyncIterator` per accident here? For whatever reason it's only present in v6 so maybe it's anyway wrong.

Reply:
I removed it intentionally. It seemed out of place and I verified that it does not, in fact, exist in Node v6.